pinot-spi/src/main/java/org/apache/pinot/spi/data/DataTable.java

```diff
@@ -33,7 +32,7 @@
   // TODO: remove this when we stop supporting DataTable V2.
   String EXCEPTION_METADATA_KEY = "Exception";
 
-  void addException(ProcessingException processingException);
+  void addException(Exception processingException);
```

Comment:
Shall we consider make
```
public interface DataTable<E> {
...
void addException(E processingException);
```

Reply:
yes. at first I thought i would've been weird to add a generic for the exception. But now that I think about it. the exception is part of the data transferred between pinot components. it make sense to genericfy the exception type to be handled.

Reply:
actually my bad. it doesn't work since most of the place doesn't rely on datatable to have type erasure. so I added the generic exception handling instead.